### PR TITLE
#162949066 Configure Authors Haven to use PostgreSQL as DB Engine

### PR DESCRIPTION
**Pull request by waracci**
### What does this PR do?

Configures the application to use Postgres as the db engine

### Description of tasks to be completed?

Configure Authors Haven to use **PostgreSQL** as DB Engine instead of **SQLite**

### How should this be manually tested?

- Install or make sure PostgresDB is on your computer
- Clone this repo git clone https://github.com/andela/ah-robotics.git
- Checkout to the Chore branch `git checkout ch-change-db-postgres-162949066`
- Create a virtual environment with the command
> `virtualenv -p python3 env`
- Create a .env file and add the following db configurations:

```
DB_NAME='database_name'
DB_USER='created_username'
DB_PASS='password_for_created_user'
DB_HOST='host_name_eg_localhost'
DB_PORT='port_number_eg_5432'
```
Source the environment variables 
> `source .env`

#### Setting Up the Database

Start Postgres Database server 
> `pg_ctl -D /usr/local/var/postgres start`
To open database on terminal use `psql Postgres`
Create a database with name `author_haven`
Migrate changes with `python manage.py makemigrations` then `python manage.py migrate`

Relevant Pivotal Tracker Stories

[#162948832](https://www.pivotaltracker.com/story/show/162949066)

screenshot 2019-01-08 at 14 27 46

> Database Configuration on Settings.py

screenshot 2019-01-08 at 14 28 00

> Created .env_example file

screenshot 2019-01-08 at 14 28 20

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -91,3 +91,6 @@ ENV/
 
 # SQLite3
 db.sqlite3
+
+# Vsode editor files
+/.vscode
```
Comment (sharkdevs) [inline .gitignore:96]: Your commit Message is not describing the action done here. What you did was actually ignoring the vscode generated files. Let the message be precise.

Comment (waracci) [reply .gitignore:96]: Well noted.
I squashed the commits to include a more descriptive commit message.

Comment (sharkdevs) [reply .gitignore:96]: Great, resolved.

**Comment by jomasim** [comment]
You can consider using dj-database_url to configure your database variables 
checkout out this link- https://pypi.org/project/dj-database-url/

**Comment by Paulstar200** [comment]
Be sure to include the name of the branch we are to check out to in your Pull Request.

**Review by Paulvitalis200 — APPROVED** [review]

**Review by Paulvitalis200 — CHANGES_REQUESTED** [review]
Be sure to include the name of the branch we are to check out to in your Pull Request.

**Review by johnwayodi — APPROVED** [review]
Followed the steps, works as expected as the tables have been set up in my database.

However, as @Paulstar200 noted, indicating your branch on the PR testing instructions will provide more clarity to the tester.

**Comment by waracci** [comment]
> You can consider using dj-database_url to configure your database variables
> checkout out this link- https://pypi.org/project/dj-database-url/

I will consider this @jomasim

**Comment by waracci** [comment]
@Paulstar200 and @johnwayodi well noted, let me update the PR to include the name of the branch to checkout to. 👍

**Comment by sharkdevs** [comment]
On the **setup database** part in the PR documentation, add the exact command to create a database instead of saying _create database ..._

**Review by Paulvitalis200 — APPROVED** [review]

**Pull request merged**